### PR TITLE
Replace subcommands with cobra

### PR DESCRIPTION
**Pull request by yokaze**
This PR replaces subcommands with cobra.
- Use `context.Background()` where originally a context from subcommands is provided.
- Use POSIX/GNU-style options; `-config` becomes `--config`.

Signed-off-by: Daichi Sakaue <daichi-sakaue@cybozu.co.jp>

### CODE REVIEW

File: pkg/etcdpasswd/cmd/user_add.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"context"
+
+	"github.com/cybozu-go/etcdpasswd"
+	"github.com/spf13/cobra"
+)
+
+var userAddConfig struct {
+	displayName string
+	group       string
+	groups      commaStrings
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/user_add.go:13]: Why don't you use `[]string` and `plag.StringSlice` or `pflag.StringVar`?

File: pkg/etcdpasswd/cmd/cli.go [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
 )
 
-var client etcdpasswd.Client
-
-// Setup setups this package.
-func Setup(c etcdpasswd.Client) {
-	client = c
-}
-
 type commaStrings []string
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/cli.go:7]: I don't think this is worth it.  See my comment elsewhere.

File: pkg/etcdpasswd/cmd/get.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+var getCmd = &cobra.Command{
+	Use:   "get CONFIG",
+	Short: "get configurations",
+	Long: `Usage: etcdpasswd get CONFIG
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/get.go:14]: This line looks spurious.

File: pkg/etcdpasswd/cmd/set.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	"strings"
+
+	"github.com/cybozu-go/etcdpasswd"
+	"github.com/spf13/cobra"
+)
+
+var setCmd = &cobra.Command{
+	Use:   "set CONFIG VALUE",
+	Short: "set configuration",
+	Long: `Usage: etcdpasswd set CONFIG VALUE
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/set.go:16]: ditto

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: pkg/etcdpasswd/cmd/user_remove.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+var userRemoveCmd = &cobra.Command{
+	Use:   "remove NAME",
+	Short: "remove an existing user",
+	Long:  "remove an existing user.",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		name := args[0]
+		return client.RemoveUser(context.Background(), name)
```
Comment (zoetrope) [inline pkg/etcdpasswd/cmd/user_remove.go:16]: You can use `cmd.Context()` instead of `context.Backgroud()`

**Review by zoetrope — COMMENTED** [review]
reviewed

**Review by zoetrope — DISMISSED** [review]
LGTM

File: pkg/etcdpasswd/cmd/get.go (hunk 3 above)
Comment (yokaze) [reply pkg/etcdpasswd/cmd/get.go]: I could not understand what spurious means here for.
Tried to use another expression, but please notify me if it is wrong.

Comment (ymmt2005) [reply pkg/etcdpasswd/cmd/get.go]: The total line should be removed as cobra will display this.

File: pkg/etcdpasswd/cmd/user_add.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	Short: "add a new user",
+	Long:  "add a new user.",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		name := args[0]
+		user := &etcdpasswd.User{
+			Name:        name,
+			DisplayName: userAddConfig.displayName,
+			Group:       userAddConfig.group,
+			Groups:      []string(userAddConfig.groups),
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/user_add.go:26]: No cast is necessary.

File: pkg/etcdpasswd/cmd/user_update.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+		}
+
+		if len(userUpdateConfig.displayName) > 0 {
+			user.DisplayName = userUpdateConfig.displayName
+		}
+		if len(userUpdateConfig.group) > 0 {
+			user.Group = userUpdateConfig.group
+		}
+		if len(userUpdateConfig.groups) > 0 {
+			user.Groups = []string(userUpdateConfig.groups)
```
Comment (ymmt2005) [inline pkg/etcdpasswd/cmd/user_update.go:34]: ditto

File: pkg/etcdpasswd/cmd/get.go (hunk 3 above)
Comment (yokaze) [reply pkg/etcdpasswd/cmd/get.go]: Is it better to remove the line entirely?
I changed the expression and now the help looks as follows.

```
$ ./etcdpasswd get --help
get configurations.

CONFIG is one of:
    start-uid:      the beginning UID assigned to managed users.
    start-gid:      the beginning GID assigned to managed users.
    default-group:  default primary group.
    default-groups: comma-separated list of supplementary groups.
    default-shell:  default shell program.

Usage:
  etcdpasswd get CONFIG [flags]

Flags:
  -h, --help   help for get

Global Flags:
      --config string      configuration file path (default "/etc/etcdpasswd/config.yml")
      --logfile string     Log filename
      --logformat string   Log format [plain,logfmt,json]
      --loglevel string    Log level [critical,error,warning,info,debug]
```

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**